airflow/utils/python_virtualenv_script.jinja2

```diff
@@ -16,7 +16,9 @@
  specific language governing permissions and limitations
  under the License.
 -#}
-
+{% if import_annotations | default(False) %}
+from __future__ import annotations
```

Comment:
Can't we just add `from __future__ import annotations` always ? I don't think it has any serious side-effects, it should be fully backwards compatible and it would remove the need of checking for annotations import in callable (which also is not a good one because I think it does not handle the case where `from __future__ import annotation` is at the top-level of the DAG (which will be actually most of the cases when someone uses is).

Reply:
I wasn't sure about the side effects, so this is how I handled it.
I don't know if this will actually happen, but it could cause an error.

```python
def annotation_error(value: int = 1) -> None:
    import inspect

    sig = inspect.signature(annotation_error)
    maybe_int = sig.parameters["value"].annotation
    print(repr(maybe_int))
    some_int = maybe_int(1)
    print(some_int)


def with_import_annotaions() -> None:
    import inspect
    import subprocess
    import sys

    func = inspect.getsource(annotation_error)
    py_source = f"""
from __future__ import annotations
{func}

if __name__ == "__main__":
    assert callable(annotation_error)
    annotation_error()
"""

    # error
    subprocess.run([sys.executable, "-c", py_source], check=True)

if __name__ == '__main__':
    annotation_error()
    # output:
    # <class 'int'>
    # 1
    with_import_annotaions()
    # output:
    # 'int'
    # TypeError: 'str' object is not callable
```

Of course, it's much easier not to consider these cases.
It's also unlikely that this is the case.
(However, I've seen similar issues with `pydantic`).

Considering that the implementation is much simpler and will be the default in the next Python version, I think it's better to change it as you said.

I'll make it as soon as possible, and fix the test code that caused the error.